tinkerpop3/src/main/java/io/shiftleft/overflowdb/OdbNode.java

```diff
@@ -448,10 +463,13 @@ private int storeAdjacentNode(Direction direction,
       }
     }
 
+    /* marking as dirty *after* we updated - if node gets serialized before we finish, it'll be marked as dirty */
+    this.markAsDirty();
```

Comment:
@mpollmeier There is also `setEdgeProperty` which is called when you set edge property from this fn and also from the OdbEdge through `property` method. I think it should also mark the node as dirty.

Reply:
we do that already: https://github.com/ShiftLeftSecurity/overflowdb/blob/3fe462229187fd5a0ae908d1f501c437745c62a1/tinkerpop3/src/main/java/io/shiftleft/overflowdb/OdbNode.java#L228